src/Maths/Silk.NET.Maths/Simd/Simd128.GreaterThan.cs

```diff
@@ -0,0 +1,208 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#if INTRINSICS
+using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+#if SSE
+using System.Runtime.Intrinsics.X86;
+#endif
+#if AdvSIMD
+using System.Runtime.Intrinsics.Arm;
+#endif
+
+namespace Silk.NET.Maths
+{
+    public static unsafe partial class Simd128
+    {
+        /// <summary>
+        /// Performs hardware-accelerated GreaterThan on 128-bit vectors.
+        /// </summary>
+        [MethodImpl(Scalar.MaxOpt)]
+        public static Vector128<T> GreaterThan<T>(Vector128<T> left, Vector128<T> right) where T : unmanaged
+        {
+            return Byte(left, right);            
+            [MethodImpl(Scalar.MaxOpt)]
+            static Vector128<T> Byte(Vector128<T> left, Vector128<T> right)
+            {
+                if (typeof(T) == typeof(byte))
+                {
+#if AdvSIMD
+                    if (AdvSimd.IsSupported)
+                    {
+                        return AdvSimd.CompareGreaterThan(left.AsByte(), right.AsByte()).As<byte, T>();
+                    }
+#endif
+#if SSE
+                    if (Sse2.IsSupported)
+                    {
+                        var leftShifted = Simd128.Add(left.AsByte(), Simd128<byte>.MaxValueOver2);
+                        var rightShifted = Simd128.Add(right.AsByte(), Simd128<byte>.MaxValueOver2);
+                        return Sse2.CompareGreaterThan(leftShifted.AsSByte(), rightShifted.AsSByte()).As<SByte, T>();
+                    }
```

Comment:
Confused. `byte` is unsigned, goes from 0 to 255. You only need the + trick for signed stuff?

Reply:
Seems to be [working](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7NAExAGoAfAAQCYAGAWAChqBGRimgAguYDoAlfBgCWAWxg8AwhBEAHIQBsYUAMpKAbkLAwAzgG5GbZgE4AFAEltAcVgBDDEpPMaNNByQBWAJSf9TY+atbeygTAGYqVw9vX25TC2sYOwdmdyRIrx9DOMDE4MdU12ZQmmisgISkkKKXDhSkUr9sirzmJBrWkszG8qCHYsKADk7fQyQOYAgIeQ543pDgAE97DhtXReXgT0YAb0YOfY41GygV5QALIQAze0IOAF4OE3WYTxMbDhJxpfEAWRsEABqNnkOBgHAA9BwqF0DodjuNzlcbvdHs9XsAPl97Dw/oDgaCIVCYQcKAB2DgAVTw2hsl3EAEFtAAeZ6ubTPAB8Jlgl1OF2uMEIng4HMp1NpDOZACFvmzOdyYLzgIiBULfABfIA)

Reply:
[More detailed version](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7NAExAGoAfAWACgABAJgEZqaAGAAhoYDoAlfDAJYBbGFwDCEIQAcBAGxhQAygoBuAsDADOAbmrMGATgAUASU0BxWAEMMCowxYs0bJAFYAlO921Dpi9dsoIwBmOmc3T29OYzNLGBs7BlckcI8vfRj/eMD7ZOcGYJZIjL84hKCCpzYkpGKfTLKchiQq5qL0+tKAu0L8gA52730kNmAICFk2WO6g4ABPWzYrZ3nF4HdqAG9qNl2OXwASACIAHTxNqwBfNhVNNk3gS6OOvZurKCXFAAsBADNbQhsAC8bCMqxg7iMVjYJFGC1EAFkrAgAGpWWQ4GBsAD0bDoLz20SMxys3z+AJA91JP3+MEITwJuxU71GZNpgJBYPhkOAMLhti4SNR6MxOLxjP2xmOwDZFPuMppAIZ3lezI+pIEAHM8HTgWwAKp4TRWX6iACCmgAPODnJpwQA+IywX6fRV0iVEkmKLU64hU73aunKnZ7NWsn26kGG42mrgWy0AIXhtodTpgLoV5PdKsJhyOMojfoeAd9waorxoAHZPoW2Pbw4HCN5LkA)